plugin-server/src/cdp/hog-executor.ts

```diff
@@ -0,0 +1,294 @@
+
+        // Filter all functions based on the invocation
+        functions = Object.fromEntries(
+            Object.entries(functions).filter(([_key, value]) => {
+                try {
+                    const filters = value.filters
+
+                    if (!filters?.bytecode) {
+                        // NOTE: If we don't have bytecode this indicates something went wrong.
+                        // The model will always safe a bytecode if it was compiled correctly
+                        return false
+                    }
+
+                    const filterResult = exec(filters.bytecode, {
+                        globals: invocation.globals,
+                        timeout: 100,
+                        maxAsyncSteps: 0,
+                    })
+
+                    if (typeof filterResult.result !== 'boolean') {
+                        // NOTE: If the result is not a boolean we should not execute the function
+                        return false
+                    }
+
+                    return filterResult.result
+                } catch (error) {
+                    status.error('🦔', `[HogExecutor] Error filtering function`, {
+                        hogFunctionId: value.id,
+                        hogFunctionName: value.name,
+                        error: error.message,
+                    })
+                }
+
+                return false
+            })
+        )
+
+        if (!Object.keys(functions).length) {
+            return []
+        }
+
+        const results: HogFunctionInvocationResult[] = []
+
+        for (const hogFunction of Object.values(functions)) {
+            // Add the source of the trigger to the globals
+            const modifiedGlobals: HogFunctionInvocationContext = {
+                ...invocation.globals,
+                source: {
+                    name: hogFunction.name ?? `Hog function: ${hogFunction.id}`,
+                    url: `${invocation.globals.project.url}/pipeline/destinations/hog-${hogFunction.id}/configuration/`,
+                },
+            }
+
+            const result = await this.execute(hogFunction, {
+                ...invocation,
+                globals: modifiedGlobals,
+            })
+
+            results.push(result)
+        }
+
+        return results
+    }
+
+    /**
+     * Intended to be invoked as a continuation from an async function
+     */
+    async executeAsyncResponse(invocation: HogFunctionInvocationAsyncResponse): Promise<any> {
+        if (!invocation.hogFunctionId) {
+            throw new Error('No hog function id provided')
+        }
+
+        const hogFunction = this.hogFunctionManager.getTeamHogFunctions(invocation.globals.project.id)[
+            invocation.hogFunctionId
+        ]
+
+        invocation.vmState.stack.push(convertJSToHog(invocation.response))
+
+        await this.execute(hogFunction, invocation, invocation.vmState)
```

Comment:
This is no longer needed --> `vmState` contains the bytecode and all else needed to resume... unless you're explicitly sanity checking the function exists?

Reply:
Yeah im leaving this quite open. For these simple functions it doesn't really matter. Once we hit bigger, longer running programs we need to consider what sort of checks would be done at each stage etc.